### PR TITLE
Replace ERR_EXPLAIN in scene/* code

### PR DESCRIPTION
**Pull request by KoBeWi**
Addresses part of #31207

Aside from straightforward adding `_MSG `, I changed some `if` conditions to `FAIL_COND_*` macros. There are however two `EXPLAIN`s I didn't know what to do about, as they come with some font-specific macro (which doesn't have `MSG` version):
https://github.com/godotengine/godot/blob/d2a67c9c1fdee470064f2b3c5750c98f174b5399/scene/resources/dynamic_font.cpp#L187-L188
(and below)

### CODE REVIEW

File: scene/2d/area_2d.cpp [hunk 1]
```diff
@@ -401,10 +398,7 @@ void Area2D::set_monitoring(bool p_enable) {
 
 	if (p_enable == monitoring)
 		return;
-	if (locked) {
-		ERR_EXPLAIN("Function blocked during in/out signal. Use set_deferred(\"monitoring\",true/false)");
-	}
-	ERR_FAIL_COND(locked);
+	ERR_FAIL_COND_MSG(locked, "Function blocked during in/out signal. Use set_deferred(\"monitoring\",true/false)");
```
Comment (akien-mga) [inline scene/2d/area_2d.cpp]: Nitpick, but while we're changing this, can you add a space after `"monitoring",`? Same below.

File: scene/2d/physics_body_2d.cpp [hunk 2]
```diff
@@ -905,10 +891,7 @@ void RigidBody2D::set_contact_monitor(bool p_enabled) {
 
 	if (!p_enabled) {
 
-		if (contact_monitor->locked) {
-			ERR_EXPLAIN("Can't disable contact monitoring during in/out callback. Use call_deferred(\"set_contact_monitor\",false) instead");
-		}
-		ERR_FAIL_COND(contact_monitor->locked);
+		ERR_FAIL_COND_MSG(contact_monitor->locked, "Can't disable contact monitoring during in/out callback. Use call_deferred(\"set_contact_monitor\",false) instead");
```
Comment (akien-mga) [inline scene/2d/physics_body_2d.cpp]: Same here.

**Comment by akien-mga** [comment]
Only checked a few of them so far, will check the rest tomorrow.

While we're editing those, might be worth make the style of error messages consistent. Some end with dots and some don't, I think it should likely always end with a dot?

**Comment by KoBeWi** [comment]
Added dots everywhere.

And what about translations? Shouldn't errors be translated? 4 out of the 100+ I changed were translated (some font-related stuff).

**Comment by akien-mga** [comment]
> And what about translations? Shouldn't errors be translated? 4 out of the 100+ I changed were translated (some font-related stuff).

Errors shouldn't be translated, they often point to engine issues and we need them in English when reported on GitHub.

File: scene/resources/dynamic_font.cpp [hunk 3]
```diff
@@ -118,8 +118,7 @@ Error DynamicFontAtSize::_load() {
 
 	int error = FT_Init_FreeType(&library);
 
-	ERR_EXPLAIN(TTR("Error initializing FreeType."));
-	ERR_FAIL_COND_V(error != 0, ERR_CANT_CREATE);
+	ERR_FAIL_COND_V_MSG(error != 0, ERR_CANT_CREATE, TTR("Error initializing FreeType."));
```
Comment (akien-mga) [inline scene/resources/dynamic_font.cpp]: The `TTR()` calls are bogus in this file, you can remove them.
`TTR` should never be used outside editor code, it gets replaced by `String()` on non-tools builds.

File: scene/animation/animation_cache.cpp [hunk 4]
```diff
@@ -80,8 +80,7 @@ void AnimationCache::_update_cache() {
 		if (!node) {
 
 			path_cache.push_back(Path());
-			ERR_EXPLAIN("Invalid Track Path in Animation: " + np);
-			ERR_CONTINUE(!node);
+			ERR_CONTINUE_MSG(!node, "Invalid Track Path in Animation: " + np + ".");
```
Comment (akien-mga) [inline scene/animation/animation_cache.cpp]: I'd lowercase "track path in animation" for style consistency, but it's a nitpick, I don't mind if it stays as is :)

File: scene/animation/tween.cpp [hunk 5]
```diff
@@ -1185,44 +1185,38 @@ bool Tween::_build_interpolation(InterpolateType p_interpolation_type, Object *p
...
-	ERR_FAIL_COND_V(!ObjectDB::instance_validate(p_object), false); // Is the object a valid instance?
+	ERR_FAIL_COND_V_MSG(p_object == NULL, false, "Invalid object provided to Tween."); // Is the object real
+	ERR_FAIL_COND_V_MSG(!ObjectDB::instance_validate(p_object), false, "Invalid object provided to Tween."); // Is the object a valid instance?
 	data.id = p_object->get_instance_id();
 
 	// Validate the initial and final values
-	ERR_EXPLAIN("Initial value type does not match final value type!"); // TODO: Print both types to make debugging easier
-	ERR_FAIL_COND_V(p_initial_val.get_type() != p_final_val.get_type(), false); // Do the initial and final value types match?
+	// TODO: Print both types to make debugging easier
+	ERR_FAIL_COND_V_MSG(p_initial_val.get_type() != p_final_val.get_type(), false, "Initial value type does not match final value type."); // Do the initial and final value types match?
```
Comment (akien-mga) [inline scene/animation/tween.cpp]: Let's solve the TODO:
```
"Initial value type '" + Variant::get_type_name(p_initial_val.get_type()) + "' does not match final value type '" + Variant::get_type_name(p_final_val.get_type()) + "'."
```
(Untested, but if it compiles it should be fine I guess.)

Comment (akien-mga) [reply scene/animation/tween.cpp]: The trailing comments in this class are a bit useless BTW, they just state exactly what the test + explanatory message say... but well, it's unrelated to this PR :P

File: scene/animation/tween.cpp [hunk 6]
```diff
@@ -1231,8 +1225,8 @@ bool Tween::_build_interpolation(InterpolateType p_interpolation_type, Object *p
 	// Is the method defined?
 	if (p_method) {
 		// Does the object even have the requested method?
-		ERR_EXPLAIN("Tween target object has no method named: " + *p_method); // TODO: Fix this error message
-		ERR_FAIL_COND_V(!p_object->has_method(*p_method), false);
+		// TODO: Fix this error message
```
Comment (akien-mga) [inline scene/animation/tween.cpp]: I don't see what's wrong with that error message, so I'd just drop the cryptic TODO.

File: scene/animation/tween.cpp [hunk 7]
```diff
@@ -1301,8 +1295,7 @@ bool Tween::interpolate_callback(Object *p_object, real_t p_duration, String p_c
 	ERR_FAIL_COND_V(p_duration < 0, false);
 
 	// Check whether the object even has the callback
-	ERR_EXPLAIN("Object has no callback named: %s" + p_callback);
-	ERR_FAIL_COND_V(!p_object->has_method(p_callback), false);
+	ERR_FAIL_COND_V_MSG(!p_object->has_method(p_callback), false, "Object has no callback named: %s" + p_callback + ".");
```
Comment (akien-mga) [inline scene/animation/tween.cpp]: These errors were (and still are) bogus. There's no substitution done. This was meant to be passed to `vformat` I guess, but here you can simply drop the `%s` (same below).

File: scene/gui/control.cpp [hunk 8]
```diff
@@ -2377,12 +2367,7 @@ Control *Control::_get_focus_neighbour(Margin p_margin, int p_count) {
 		Node *n = get_node(data.focus_neighbour[p_margin]);
 		if (n) {
 			c = Object::cast_to<Control>(n);
-
-			if (!c) {
-
-				ERR_EXPLAIN("Neighbour focus node is not a control: " + n->get_name());
-				ERR_FAIL_V(NULL);
-			}
+			ERR_FAIL_COND_V_MSG(!c, NULL, "Neighbour focus node is not a control: " + n->get_name() + ".");
```
Comment (akien-mga) [inline scene/gui/control.cpp]: en_GB spotted, should be `Neighbor`.

File: scene/main/node.cpp [hunk 9]
```diff
@@ -1239,10 +1218,7 @@ void Node::_propagate_validate_owner() {
 void Node::remove_child(Node *p_child) {
 
 	ERR_FAIL_NULL(p_child);
-	if (data.blocked > 0) {
-		ERR_EXPLAIN("Parent node is busy setting up children, remove_node() failed. Consider using call_deferred(\"remove_child\",child) instead.");
-		ERR_FAIL_COND(data.blocked > 0);
-	}
+	ERR_FAIL_COND_MSG(data.blocked > 0, "Parent node is busy setting up children, remove_node() failed. Consider using call_deferred(\"remove_child\",child) instead.");
```
Comment (akien-mga) [inline scene/main/node.cpp]: And space after `"remove_child",`.

File: scene/resources/text_file.cpp [hunk 10]
```diff
@@ -67,10 +67,7 @@ Error TextFile::load_text(const String &p_path) {
 	w[len] = 0;
 
 	String s;
-	if (s.parse_utf8((const char *)w.ptr())) {
-		ERR_EXPLAIN("Script '" + p_path + "' contains invalid unicode (utf-8), so it was not loaded. Please ensure that scripts are saved in valid utf-8 unicode.");
-		ERR_FAIL_V(ERR_INVALID_DATA);
-	}
+	ERR_FAIL_COND_V_MSG(s.parse_utf8((const char *)w.ptr()), ERR_INVALID_DATA, "Script '" + p_path + "' contains invalid unicode (utf-8), so it was not loaded. Please ensure that scripts are saved in valid utf-8 unicode.");
```
Comment (akien-mga) [inline scene/resources/text_file.cpp]: While I'm nitpicking, maybe capitalize "UTF-8" here? (two occurrences)

**Comment by akien-mga** [comment]
Checked everything, seems good. Left a few nitpicks, once solved it can be merged.

File: scene/gui/control.cpp (hunk 8 above)
Comment (KoBeWi) [reply scene/gui/control.cpp]: The method itself has "neighbour" in name...

File: scene/animation/tween.cpp (hunk 5 above)
Comment (KoBeWi) [reply scene/animation/tween.cpp]: I was going to solve it, but wasn't sure how to get the type string.
(also removed the comments xd)

**Comment by KoBeWi** [comment]
I resolved the nitpicks.

File: scene/gui/control.cpp (hunk 8 above)
Comment (akien-mga) [reply scene/gui/control.cpp]: Oops :D Probably one to list in #16863.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**